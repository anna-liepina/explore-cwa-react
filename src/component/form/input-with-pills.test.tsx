import React from 'react';
import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import InputWithPills from './input-with-pills';

describe('<InputWithPills/>', () => {
    const props = {
    };

    const optionalProps = {
        value: [
            {
                label: 'tag 0 - value as string',
                value: 'string'
            },
            {
                label: 'tag 1 - value as number',
                value: 1
            }
        ],
        'data-cy': 'optProps.data-cy',
        className: 'optProps.className',
    };

    describe('render', () => {
        it('with default/required props', () => {
            const { asFragment } = render(<InputWithPills {...props} />);

            expect(asFragment()).toMatchSnapshot();
        });

        it('with optional/required props', () => {
            const { asFragment } = render(<InputWithPills {...props} {...optionalProps} />);

            expect(asFragment()).toMatchSnapshot();
        });
    });
});
