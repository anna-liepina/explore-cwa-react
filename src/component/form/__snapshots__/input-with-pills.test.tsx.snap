// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputWithPills/> render with default/required props 1`] = `
<DocumentFragment>
  <div
    class="input-with-pills"
  >
    <input
      class="input-with-pills__input"
      data-cy=""
    />
  </div>
</DocumentFragment>
`;

exports[`<InputWithPills/> render with optional/required props 1`] = `
<DocumentFragment>
  <div
    class="input-with-pills optProps.className"
  >
    <span
      class="pill"
      data-cy="optProps.data-cy-pill-0"
    >
      tag 0 - value as string
      <button
        class="pill__control"
        data-cy="optProps.data-cy-pill-0-remove"
        data-id="0"
      >
        ×
      </button>
    </span>
    <span
      class="pill"
      data-cy="optProps.data-cy-pill-1"
    >
      tag 1 - value as number
      <button
        class="pill__control"
        data-cy="optProps.data-cy-pill-1-remove"
        data-id="1"
      >
        ×
      </button>
    </span>
    <input
      class="input-with-pills__input"
      data-cy="optProps.data-cy"
    />
  </div>
</DocumentFragment>
`;
